tests/core/cohere/test_call_response.py

```diff
@@ -0,0 +1,148 @@
+"""Tests the `cohere.call_response` module."""
+
+from cohere.types import (
+    ApiMeta,
+    ApiMetaBilledUnits,
+    ChatMessage,
+    NonStreamedChatResponse,
+    ToolCall,
+    ToolResult,
+)
+
+from mirascope.core.cohere.call_response import CohereCallResponse
+from mirascope.core.cohere.tool import CohereTool
+
+
+def test_cohere_call_response() -> None:
+    """Tests the `CohereCallResponse` class."""
+    usage = ApiMetaBilledUnits(input_tokens=1, output_tokens=1)
+    completion = NonStreamedChatResponse(
+        generation_id="id",
+        text="content",
+        finish_reason="COMPLETE",
+        meta=ApiMeta(billed_units=usage),
+    )
+    call_response = CohereCallResponse(
+        metadata={},
+        response=completion,
+        tool_types=None,
+        prompt_template="",
+        fn_args={},
+        dynamic_config=None,
+        messages=[],
+        call_params={},
+        call_kwargs={},
+        user_message_param=None,
+        start_time=0,
+        end_time=0,
+    )
+    call_response._model = "command-r-plus"
+    assert call_response._provider == "cohere"
+    assert call_response.content == "content"
+    assert call_response.finish_reasons == ["COMPLETE"]
+    assert call_response.model == "command-r-plus"
+    assert call_response.id == "id"
+    assert call_response.usage == usage
+    assert call_response.input_tokens == 1
+    assert call_response.output_tokens == 1
+    assert call_response.cost == 1.8e-5
+    assert call_response.message_param == ChatMessage(
+        message="content",
+        role="assistant",  # type: ignore
```

Comment:
you can also resolve this but ugh not having role as a key is the weirdest most annoying thing like why

Reply:
story of all my tests